source/openapi-admin-v3.yaml

```diff
@@ -4734,152 +4728,235 @@ components:
           description: |-
             A list of valid partition keys based on the cluster's collection
             schemas.
-    TriggerResponse:
-      properties:
-        _id:
-          type: string
-          description: The unique ID of the trigger.
-        name:
-          type: string
-          description: The name of the trigger.
-        type:
-          type: string
-          description: The type of the trigger.
-        function_id:
-          type: string
-          description: The ID of the function associated with the trigger.
-        function_name:
-          type: string
-          description: The name of the function associated with the trigger.
-        disabled:
-          type: boolean
-          description: If true, the trigger is disabled and will not fire.
-    TriggerRequest:
+    AnyTrigger:
+      oneOf:
+        - $ref: "#/components/schemas/DatabaseTrigger"
+        - $ref: "#/components/schemas/AuthenticationTrigger"
+        - $ref: "#/components/schemas/ScheduledTrigger"
+    Trigger:
+      oneOf:
+        - $ref: "#/components/schemas/DatabaseTrigger"
+        - $ref: "#/components/schemas/AuthenticationTrigger"
+        - $ref: "#/components/schemas/ScheduledTrigger"
+      discriminator:
+        propertyName: type
```

Comment:
Is this mapping only possible in a property? It took me a minute to figure out that switching the type here changed the configuration value. The `config` object is collapsed by default, so the only change I saw when selecting a different option in the drop-down was the "this value is always "[FOO]"." changing in the `type` description. If we could provide the mapping drop-down in the "Request Body Schema" description - i.e. `A configuration for a database, scheduled, or authentication trigger.` - I think it might be more intuitive to select it there and understand that the subsequent body was changing, even if you couldn't see the config.

Separately, I was also a little puzzled when I saw that the request sample example had a different mapping drop-down that I had to select independent of selecting it here. Is it possible to have the example body derive from the selection here?

Reply:
Nice - I think this UX makes the expected result a little more apparent. I would love to see the example body be linked, but at least now with the pills it seems a little less like recreating the same action and for some reason, that makes it slightly less annoying to have to do it separately.